shared/uavobjectdefinition/hwsprf3e.xml

```diff
@@ -0,0 +1,142 @@
+				<option>Disabled</option>
+				<option>DSM</option>
+				<option>SRXL</option>
+				<option>HoTT SUMD</option>
+				<option>HoTT SUMH</option>
+				<option>S.Bus</option>
+				<option>S.Bus Non Inverted</option>
+				<option>IBus</option>
+				<option>ComBridge</option>
+				<option>DebugConsole</option>
+				<option>FrSKY Sensor Hub</option>
+				<option>FrSKY SPort Telemetry</option>
+				<option>GPS</option>
+				<option>HoTT Telemetry</option>
+				<option>I2C</option>
+				<option>LighttelemetryTx</option>
+				<option>MavLinkTX</option>
+				<option>MavLinkTX_GPS_RX</option>
+				<option>MSP</option>
+				<option>OpenLog</option>
+				<option>Storm32Bgc</option>
+				<option>Telemetry</option>
+				<option>TBS Crossfire</option>
+			</options>
+		</field>
+		<field name="Uart2" units="function" type="enum" elements="1" parent="HwShared.PortTypes" defaultvalue="S.Bus">
+			<description/>
+			<options>
+				<option>Disabled</option>
+				<option>DSM</option>
+				<option>SRXL</option>
+				<option>HoTT SUMD</option>
+				<option>HoTT SUMH</option>
+				<option>PPM</option>
+				<option>S.Bus</option>
+				<option>S.Bus Non Inverted</option>
+				<option>IBus</option>
+				<option>ComBridge</option>
+				<option>DebugConsole</option>
+				<option>FrSKY Sensor Hub</option>
+				<option>FrSKY SPort Telemetry</option>
+				<option>GPS</option>
+				<option>HoTT Telemetry</option>
+				<option>I2C</option>
+				<option>LighttelemetryTx</option>
+				<option>MavLinkTX</option>
+				<option>MavLinkTX_GPS_RX</option>
+				<option>MSP</option>
+				<option>OpenLog</option>
+				<option>Storm32Bgc</option>
+				<option>Telemetry</option>
+				<option>TBS Crossfire</option>
+			</options>
+		</field>
+		<field name="Uart3" units="function" type="enum" elements="1" parent="HwShared.PortTypes" defaultvalue="FrSKY SPort Telemetry">
+			<description/>
+			<options>
+				<option>Disabled</option>
+				<option>ComBridge</option>
+				<option>DebugConsole</option>
+				<option>DSM</option>
+				<option>SRXL</option>
+				<option>HoTT SUMD</option>
+				<option>HoTT SUMH</option>
+				<option>FrSKY Sensor Hub</option>
+				<option>FrSKY SPort Telemetry</option>
+				<option>GPS</option>
+				<option>HoTT Telemetry</option>
+				<option>LighttelemetryTx</option>
+				<option>MavLinkTX</option>
+				<option>MavLinkTX_GPS_RX</option>
+				<option>MSP</option>
+				<option>OpenLog</option>
+				<option>S.Bus</option>
+				<option>S.Bus Non Inverted</option>
+				<option>Storm32Bgc</option>
+				<option>Telemetry</option>
+				<option>IBus</option>
+				<option>TBS Crossfire</option>
+			</options>
+		</field>
+		<field name="USB_HIDPort" units="function" type="enum" elements="1" parent="HwShared.USB_HIDPort" defaultvalue="USBTelemetry">
+			<description/>
+		</field>
+		<field name="USB_VCPPort" units="function" type="enum" elements="1" parent="HwShared.USB_VCPPort" defaultvalue="Disabled">
+			<description>Virtual COM Port protocol</description>
+			<options>
```

Comment:
I believe this whole options node can be removed.
The parent defines also MSP, which is now excluded by specifying this list.